cloud/pkg/edgecontroller/controller/downstream.go

```diff
@@ -223,85 +223,96 @@ func (dc *DownstreamController) syncEdgeNodes() {
-							msg.Content = svcs
-							if err := dc.messageLayer.Send(*msg); err != nil {
-								klog.Warningf("Send message failed with error: %s, operation: %s, resource: %s", err, msg.GetOperation(), msg.GetResource())
-							} else {
-								klog.V(4).Infof("Send message successfully, operation: %s, resource: %s", msg.GetOperation(), msg.GetResource())
-							}
-
-							for _, svc := range svcs {
-								namespace := svc.GetNamespace()
-								selector := labels.NewSelector()
-								for k, v := range svc.Spec.Selector {
-									r, _ := labels.NewRequirement(k, selection.Equals, []string{v})
-									selector.Add(*r)
-								}
-
-								pods, err := dc.podLister.Pods(namespace).List(selector)
-								if err != nil {
-									continue
-								}
-
-								msg := model.NewMessage("")
-								resource, err := messagelayer.BuildResource(node.Name, svc.Namespace, model.ResourceTypePodlist, svc.Name)
-								if err != nil {
-									klog.Warningf("Built message resource failed with error: %v", err)
-									continue
-								}
-								msg.BuildRouter(modules.EdgeControllerModuleName, constants.GroupResource, resource, model.UpdateOperation)
-								msg.Content = pods
-								if err := dc.messageLayer.Send(*msg); err != nil {
-									klog.Warningf("Send message failed with error: %s, operation: %s, resource: %s", err, msg.GetOperation(), msg.GetResource())
-								} else {
-									klog.V(4).Infof("Send message successfully, operation: %s, resource: %s", msg.GetOperation(), msg.GetResource())
-								}
-							}
-
-							// send all endpoints to edge
-							msg = model.NewMessage("")
-							// TODO: what should in place of namespace and endpoints when we are sending endpoints list ?
-							resource, err = messagelayer.BuildResource(node.Name, "namespace", common.ResourceTypeEndpointsList, "endpoints")
-							if err != nil {
-								klog.Warningf("Built message resource failed with error: %s", err)
-								break
-							}
-							msg.BuildRouter(modules.EdgeControllerModuleName, constants.GroupResource, resource, model.UpdateOperation)
-							msg.Content = dc.lc.GetAllEndpoints()
-							if err := dc.messageLayer.Send(*msg); err != nil {
-								klog.Warningf("Send message failed with error: %s, operation: %s, resource: %s", err, msg.GetOperation(), msg.GetResource())
-							} else {
-								klog.V(4).Infof("Send message successfully, operation: %s, resource: %s", msg.GetOperation(), msg.GetResource())
-							}
+					if nsc.Type != v1.NodeReady {
+						continue
+					}
+					nstatus := string(nsc.Status)
+					status, _ := dc.lc.GetNodeStatus(node.ObjectMeta.Name)
+					dc.lc.UpdateEdgeNode(node.ObjectMeta.Name, nstatus)
+					if nsc.Status != v1.ConditionTrue || status == nstatus {
+						continue
+					}
+					// send all services to edge
+					msg := model.NewMessage("").SetRoute(modules.EdgeControllerModuleName, constants.GroupResource)
+
```

Comment:
the following three msgs(`svcMsg`, `podsMsg`, `epsMsg`) all apply to `SetResourceOperation` with the same operation `model.UpdateOperation` ， so it can be applied to msg in advance
Or dont use the common msg,  new three msgs from the beginning.

Reply:
There is no direct method, it can directly use struct field, which is more appropriate？

Reply:
yes, you are right